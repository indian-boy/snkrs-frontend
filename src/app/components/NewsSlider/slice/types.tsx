import { FunctionComponent, SVGProps } from 'react';

export interface SliderItem {
  id: number;
  title: string;
  link: {
    label: string;
    url: string;
  };
  Image: FunctionComponent<
    SVGProps<SVGSVGElement> & { title?: string | undefined }
  >;
}

export interface NewsSliceState {
  selected: SliderItem;
  items: SliderItem[];
}
