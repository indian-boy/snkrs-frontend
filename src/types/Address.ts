export interface Address {
  latitude: number;
  longitude: number;
  location: string;
}
