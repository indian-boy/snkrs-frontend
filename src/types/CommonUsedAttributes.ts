export interface CommonUsedAttributes {
  'data-testid': string;
}
