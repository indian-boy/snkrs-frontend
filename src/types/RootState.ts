import { NewsSliceState } from 'app/components/NewsSlider/slice/types';
import { ThemeState } from 'styles/themes/slice/types';

// [IMPORT NEW CONTAINERSTATE ABOVE] < Needed for generating containers seamlessly

/*
  Because the redux-injectors injects your reducers asynchronously somewhere in your code
  You have to declare them here manually
*/
export interface RootState {
  theme?: ThemeState;
  newsSlice?: NewsSliceState;
  // [INSERT NEW REDUCER KEY ABOVE] < Needed for generating containers seamlessly
}
