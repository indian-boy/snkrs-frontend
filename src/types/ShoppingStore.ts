import { Address } from './Address';

export interface ShoppingStore {
  id: number;
  placeName: string;
  address: Address;
  availability: number;
  serviceHours: string;
}
