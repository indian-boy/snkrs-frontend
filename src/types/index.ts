import { Address } from './Address';
import { RootState } from './RootState';
import { ShoppingStore } from './ShoppingStore';

export type { RootState, Address, ShoppingStore };
