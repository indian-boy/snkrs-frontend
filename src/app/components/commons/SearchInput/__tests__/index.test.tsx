import { fireEvent, render } from '@testing-library/react';
import React from 'react';
import { Provider } from 'react-redux';
import { Store } from 'redux';
import { configureAppStore } from 'store/configureStore';
import { CommonUsedAttributes } from 'types';
import { SearchInput } from '..';

jest.mock('react-i18next', () => ({
  useTranslation: () => {
    return {
      t: str => str,
      i18n: {
        changeLanguage: () => new Promise(() => {}),
      },
    };
  },
}));

const renderComponentWithProviders = (
  props: Parameters<typeof SearchInput>[number] & CommonUsedAttributes,
  store: Store,
) =>
  render(
    <Provider store={store}>
      <SearchInput {...props} />
    </Provider>,
  );

describe('<SearchInput />', () => {
  let store: ReturnType<typeof configureAppStore>;

  beforeEach(() => {
    store = configureAppStore();
  });

  it('should match snapshot', () => {
    const { getByTestId, container } = renderComponentWithProviders(
      {
        'data-testid': 'searchID',
        type: 'search',
        placeholder: 'placeholder',
      },
      store,
    );

    const searchElement = getByTestId('searchID');

    expect(searchElement).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should clear search', () => {
    const { getByTestId, getByRole } = renderComponentWithProviders(
      {
        'data-testid': 'searchID',
        type: 'search',
        placeholder: 'placeholder',
        showCloseIcon: true,
        showSearchIcon: true,
      },
      store,
    );

    const searchElement = getByTestId('searchID') as HTMLInputElement;
    const closeIcon = getByRole('clear');

    fireEvent.click(closeIcon);

    fireEvent.change(searchElement, { target: { value: 'test value' } });
    expect(searchElement.value).toBe('test value');

    fireEvent.click(closeIcon);

    expect(searchElement.value).toBe('');
  });

  it('should hide input', () => {
    const { queryByTestId } = renderComponentWithProviders(
      {
        'data-testid': 'searchID',
        type: 'search',
        hidden: true,
      },
      store,
    );

    const searchElement = queryByTestId('searchID') as HTMLInputElement;
    expect(searchElement).toBeFalsy();
  });
});
