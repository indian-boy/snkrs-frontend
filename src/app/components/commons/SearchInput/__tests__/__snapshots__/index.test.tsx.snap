// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchInput  /> should match snapshot 1`] = `
.c0 {
  position: relative;
}

.c1 {
  font-size: 1rem;
  padding: 0.9rem 2.5rem 0.9rem 0.5rem;
  outline: none;
  background-color: #FFFFFF;
  border: 1px solid #D6D6D6;
  color: #737373;
}

.c1::-webkit-search-decoration,
.c1::-webkit-search-cancel-button,
.c1::-webkit-search-results-button,
.c1::-webkit-search-results-decoration {
  display: none;
}

.c1::-webkit-input-placeholder {
  color: #737373;
  opacity: 1;
}

.c1::-moz-placeholder {
  color: #737373;
  opacity: 1;
}

.c1:-ms-input-placeholder {
  color: #737373;
  opacity: 1;
}

.c1::placeholder {
  color: #737373;
  opacity: 1;
}

.c1:-ms-input-placeholder {
  color: #737373;
}

<label
  class="c0"
>
  <input
    class="c1"
    type="search"
  />
</label>
`;
